lib/assessment.rb

```diff
@@ -5,36 +5,52 @@ class Assessment
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
-    # Your code goes here
+    enum.max {|a, b| a.length <=> b.length}
+    end
   end
 
+  puts longest
+
   # Question 2
   # Write ruby code to return the sum of entries in the
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.reduce{|a, b| a + b}
+    end
   end
 
+  puts sum
+
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the numbers enumerable are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.any? {|number| number%2 == 1}
+    end
   end
 
+  puts some_odd
+
   # Question 4
   # Write ruby code that returns true if every entry in the
   # numbers enumerable are even
   def every_even(numbers)
-    # Your code goes here
+    numbers.all? {|number| number%2 == 0}
+    end
   end
 
+
+
   # Question 5
   # Write ruby code that returns an array with all of the
   # entries from the words enumerable capitalized.
+
+  #help
   def transform(words)
-    # Your code goes here
+    memo = 0
+    upcase.map {|memo, word| memo + word
+    end
```

Comment:
You're thinking along the right lines, but you've put them together incorrectly. When you use `.map`, what gets transformed? What determines how the transformation works?

Also, are you sure `.upcase` is the right tool to use if we just want the words to be capitalized (as opposed to being in all-caps)?